index.bs

```diff
@@ -753,17 +753,20 @@ to display an overlay which informs the user that they have entered fullscreen,
 and to advertise a simple mechanism to exit fullscreen (typically the `Esc` key).
 </div>
 
-<h3 class=question id="non-fully-active">
-  How does your feature handle non-"fully active" documents?
+<h3 class=question id="bfcache">
+  What happens when a document that uses your feature is kept alive in BFCache
+  (instead of getting destroyed) after navigation, and potentially gets reused
+  on future navigations back to the document?
 </h3>
 
 After a user navigates away from a document,
-the document might stay around in a non-"[=Document/fully active=]" state,
+the document might stay around in a non-"[=Document/fully active=]" (~BFCached) state,
```

Comment:
This makes the sentence look like as if non-fully active means bfcache. But that is usually not the case, given iframes, as an example.

Reply:
Hmm yeah I was wondering about that too here and in the Design Principles version. I want to use "fully active" more often than "BFCache" because it's an actual spec term but it's not equivalent as you said. Would it be enough to just add a note saying so like in the [BFCache Guide](https://rakina.github.io/bfcache-guide/#intro), or do you have other suggestions?

Reply:
Hmm, saying what exactly?

Reply:
So I think instead of using `non-"fully active" (BFCached) document` I would instead write `non-"fully active" BFCached document` and rewrite things a bit, and also add this note below

> Note: It is possible for a document to become non-fully active for other reasons not related to BFcaching, such as when the iframe holding the document gets detached. This question only focuses on the case where BFCache is involved, and not other cases.

(Updated the PRs now with the rewritten version, PTAL!)

Reply:
Friendly ping on this, can you PTAL again @smaug---- ? :)

Reply:
I don't quite understand what 
 `non-"fully active" BFCached document` is trying to capture.
Is there a such thing as active bfcached document? Or is it trying to refer to a document which isn't fully active but is bfcached? (but bfcached is never fully active.)

Reply:
Ah OK, I was trying to use both the term that is used in spec (fully active) and the more common "BFCached" term which isn't actually used in the spec. I guess we can just use "BFCached document" for everything but have a note that mention for spec text use the term non-"fully active" (while being aware that there are other cases where a document can become non-fully active)

Reply:
Hmm, 

> for spec text use the term non-"fully active"

why that? non-fully active isn't exactly a super set of bfcached. JS isn't suppose to be able to modify bfcached document, but it can still touch some other non-fully active documents.

Reply:
I mean functionally. JS shouldn't be able to touch BFCached documents, but it can touch documents detached from iframes.

Reply:
The point is that BFCached documents are, well, bfcached. They aren't fully active but they have also other behaviors. Then there are some other non-fully active document which have different behavior.

One certainly can access DOM of detached iframes, if you take reference to the DOM before removal.  Or one may have taken reference to some other objects living in the relevant global. 
But yes, there are certainly operations (like some sort of updates) which shouldn't happen in any non-fully active document.

Reply:
Yeah it definitely would be nicer if we have a spec term just for BFCache. I guess there's also the Page Lifecycle State API's "frozen" state but that's not part of the HTML spec (I think?) and would also cover other cases. 

@domenic @annevk WDYT of adding "is cached" state instead of recommending everyone to check for "not fully active"? I think the spec addition will be basically be something like. "A document can be cached if it's not the active document and...[not sure what the extra condition that makes it differ from detached iframes... can we check that it's a top-level document?]. If a document is cached, it is not fully active."

Reply:
@annevk said
> I think we need to keep this question and add a question around BFCache.
> Whether we should add state to documents specifically for BFCache depends on whether APIs will need that  information. I suspect most will just need to care for "fully active" and the potential for that to change over time.

This is kinda how I feel as well, I think right now there is no difference in how we treat the two cases, but maybe in the future it might be different (although unlikely)? e.g. maybe some APIs might want to include documents from detached iframes (which doesn't follow [this advice](https://w3ctag.github.io/design-principles/#omit-non-fully-active) for BFCache), and that's observable if we have a reference to something in the detached document like @smaug---- mentioned
Just want to make sure we don't have to change these documentations again.....

Reply:
> @domenic @annevk WDYT of adding "is cached" state instead of recommending everyone to check for "not fully active"?

I would be against that. I think specs and APIs should use the "fully active" predicate, and not distinguish between bfcached documents and detached iframes.

I think @smaug---- has a good point that the technical language here is not precise, but I don't think we should create separate precise terms. We should just explain the situation. Something like what you already have, but instead of 

> This guideline only focuses on the case where BFCache is involved, and not other cases that might cause a document to become non-[=Document/fully active=].

I would say

> Our advice is that all non-fully active documents should be treated the same way. The only difference is that bfcached documents might become fully active again, whereas documents in detached iframes will stay inactive forever. Thus, we suggest paying extra attention to the bfcache case.

Reply:
Thanks for confirming, I've changed the note to be as suggested!